ultradns/zone.go

```diff
@@ -0,0 +1,234 @@
+	LastRefreshStatus        string `json:"lastRefreshStatus"`
+	LastRefreshStatusMessage string `json:"lastRefreshStatusMessage"`
+}
+
+type ZoneResponse struct {
+	Properties *ZoneProperties `json:"properties"`
+
+	//Primary Zone Response
+	RegistrarInfo   *RegistrarInfo   `json:"registrarInfo,omitempty"`
+	Tsig            *Tsig            `json:"tsig,omitempty"`
+	RestrictIPList  []*RestrictIp    `json:"restrictIpList,omitempty"`
+	NotifyAddresses []*NotifyAddress `json:"notifyAddresses,omitempty"`
+
+	//Secondary Zone Response
+	PrimaryNameServers       *PrimaryNameServers    `json:"primaryNameServers,omitempty"`
+	TransferStatusDetails    *TransferStatusDetails `json:"transferStatusDetails,omitempty"`
+	NotificationEmailAddress string                 `json:"notificationEmailAddress,omitempty"`
+
+	//Alias Zone Response
+	OriginalZoneName string `json:"originalZoneName,omitempty"`
+}
+
+type ZoneListResponse struct {
+	QueryInfo  *QueryInfo      `json:"queryInfo"`
+	ResultInfo *ResultInfo     `json:"resultInfo"`
+	Zones      []*ZoneResponse `json:"zones"`
+}
+
+//create zone
+func (c *Client) CreateZone(zone Zone) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	res, err := c.Do("POST", "zones", zone, target)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		errDataList := target.Error
+		return res, fmt.Errorf("error while creating a zone (%v) - %s", zone.Properties.Name, errDataList[0])
+	}
+
+	if res.StatusCode == 202 {
+		taskId := res.Header.Get("X-Task-Id")
+		er := c.TaskWait(taskId, 5, 10)
+
+		if er != nil {
+			return res, er
+		}
+	}
+
+	return res, nil
+}
+
+//read zone
+func (c *Client) ReadZone(zoneName string) (*http.Response, *ZoneResponse, error) {
+	target := Target(&ZoneResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do("GET", "zones/"+zoneName, nil, target)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		errDataList := target.Error
+		return res, nil, fmt.Errorf("error while reading a zone (%v) - %s", zoneName, errDataList[0])
+	}
+	zoneResponse := target.Data.(*ZoneResponse)
+
+	return res, zoneResponse, nil
+}
+
+//update zone
+func (c *Client) UpdateZone(zoneName string, zone Zone) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do("PUT", "zones/"+zoneName, zone, target)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		errDataList := target.Error
+		return res, fmt.Errorf("error while updating a zone (%v) - %s", zoneName, errDataList[0])
+	}
+
+	return res, nil
+}
+
+//delete zone
+func (c *Client) DeleteZone(zoneName string) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do("DELETE", "zones/"+zoneName, nil, target)
```

Comment:
Looks like there are some existing constant for the http methods in the net/http lib: https://pkg.go.dev/net/http#pkg-constants

Please try to use those constants instead at all the places.